packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx

```diff
@@ -44,10 +48,38 @@ export const ContextMenu: React.FC<ContextMenuProps> = React.memo(({ x, y, onClo
     }
   });
 
+  const onClick = useCallback(() => {
+    if (onClose) {
+      onClose();
+    }
+  }, [onClose]);
+
   const header = renderHeader && renderHeader();
   return (
     <Portal>
-      <Menu header={header} items={items} onClose={onClose} ref={menuRef} style={positionStyles} />
+      <Menu
+        header={header}
+        ref={menuRef}
+        style={positionStyles}
+        ariaLabel={selectors.components.Menu.MenuComponent('Context')}
+      >
+        {itemsGroup?.map((group, index) => (
+          <MenuGroup key={`${group.label}${index}`} label={group.label} ariaLabel={group.label}>
+            {(group.items || []).map((item) => (
+              <MenuItem
+                key={`${item.label}`}
+                url={item.url}
+                label={item.label}
+                ariaLabel={item.label}
+                target={item.target}
+                icon={item.icon}
+                active={item.active}
+                onClick={onClick}
+              />
+            ))}
+          </MenuGroup>
+        ))}
```

Comment:
@torkelo This seems to have solved the issues of `datalinks` and `add annotations` not showing in the old graph. Figured out what caused the issue in the first place: failed to assign a unique key to both MenuGroup and MenuItem components respectively 😌

You can test it yourself.

I think the only issue I'm beginning to notice is that the `add annotation` button isn't working. Looking into now

Reply:
no that was not the source of the bug, it was here: https://github.com/grafana/grafana/pull/31639/files#diff-86bb7f4c86d169e90346c8c9e54f15e1b6d7f7a84653b61a9b0477b0fb9b1d1bL11    (messy angular wrapper for the react component)

Reply:
oh wow...interesting.
Never knew it was there. 

Thanks for letting me know Torkel

Reply:
I'm finding it a bit hard to identify the root cause of the `add annotations` issue.

I assume it must be a result of some inconsistencies during click events.

Any idea on where this issue originates?

Reply:
it was due to rename of prop items -> itemsGroups in the GraphContextMenu